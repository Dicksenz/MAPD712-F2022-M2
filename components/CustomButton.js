import React from "react";
import { View, StyleSheet, Text, TouchableOpacity } from "react-native";

const CustomButton = ({ title, onPress }) => {
  return (
    <TouchableOpacity onPress={onPress}>
      <View style={styles.container}>
        <Text style={styles.titleStyle}>{title}</Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    marginTop: 40,
    backgroundColor: "#347174",
    height: 50,
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
  },

  titleStyle: {
    fontSize: 15,
    color: "white",
  },
});

export default CustomButton;
