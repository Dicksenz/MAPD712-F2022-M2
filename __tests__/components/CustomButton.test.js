import React from "react";
import renderer from "react-test-renderer";
import CustomButton from "../../components/CustomButton";

test("Check if CustomButton component renders correctly", () => {
  const tree = renderer.create(<CustomButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
