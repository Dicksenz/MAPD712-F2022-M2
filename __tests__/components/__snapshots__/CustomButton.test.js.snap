// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if CustomButton component renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#347174",
        "display": "flex",
        "height": 50,
        "justifyContent": "center",
        "marginTop": 40,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 15,
        }
      }
    />
  </View>
</View>
`;
